### PR TITLE
RavenDB-18484  Better detection if server is secured

### PR DESCRIPTION
**Pull request by Danielle9897**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18484

### Additional description
detect if the server is secure according to the certificate

### Type of change
- Bug fix

### How risky is the change?
- Low 

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing 
- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/settings/discoveryUrl.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ﻿/// <reference path="../../../../typings/tsd.d.ts"/>
 import jsonUtil = require("common/jsonUtil");
+import accessManager = require("common/shell/accessManager");
 
 class discoveryUrl {
     discoveryUrlName = ko.observable<string>();
     validationGroup: KnockoutValidationGroup;
 
-    static usingHttps = location.protocol === "https:";
+    static usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/settings/discoveryUrl.ts]: can we rename left part as well? This is misleading now.

File: src/Raven.Studio/typescript/viewmodels/manage/certificates.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ class certificates extends viewModelBase {
     
     domainsForServerCertificate = ko.observableArray<string>([]);
     
-    usingHttps = location.protocol === "https:";
+    usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/certificates.ts]: rename prop

File: src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts [hunk 3]
```diff
@@ -66,7 +67,7 @@ type certificateInfo = {
 
 class trafficWatch extends viewModelBase {
     
-    static readonly usingHttps = location.protocol === "https:";
+    static readonly usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts]: rename prop

File: src/Raven.Studio/typescript/viewmodels/resources/createDatabase.ts [hunk 4]
```diff
@@ -43,7 +44,7 @@ class createDatabase extends dialogViewModelBase {
     clusterNodes: clusterNode[] = [];
     
     encryptionSection: setupEncryptionKey;
-    usingHttps = location.protocol === "https:"; 
+    usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/resources/createDatabase.ts]: rename

File: src/Raven.Studio/typescript/viewmodels/resources/serverDashboard.ts [hunk 5]
```diff
@@ -752,7 +752,7 @@ class serverDashboard extends viewModelBase {
     node: KnockoutComputed<clusterNode>;
     sizeFormatter = generalUtils.formatBytesToSize;
 
-    usingHttps = location.protocol === "https:";
+    usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/resources/serverDashboard.ts]: rename

File: src/Raven.Studio/typescript/viewmodels/resources/widgets/licenseWidget.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ interface serverCertificateInfo {
 class licenseWidget extends widget {
 
     refreshIntervalId: number = -1;
-    usingHttps = location.protocol === "https:";
+    usingHttps = accessManager.default.secureServer();
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/resources/widgets/licenseWidget.ts]: rename

File: src/Raven.Studio/typescript/viewmodels/shell.ts [hunk 7]
```diff
@@ -249,6 +249,7 @@ class shell extends viewModelBase {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/shell.ts:250]: call this.accessManager.secureServer(true); here.

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/shell.ts:250]: sure - can do that (done) - although it is already initialized with true in `accessManager.ts` when declared

File: src/Raven.Studio/typescript/models/database/settings/discoveryUrl.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/settings/discoveryUrl.ts]: done - rename to `isSecureServer` in all relevant places

**Review by ml054 — APPROVED** [review]

**Pull request merged**